charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml

```diff
@@ -39,10 +39,10 @@ spec:
 {{ toYaml . | indent 8 }}
 {{- end }}
     # runOnControlPlane=true or runOnMaster=true only takes effect if affinity is not set
-    {{- if contains (tpl "{{ .Values.controller.affinity }}" .) "nodeSelectorTerms" }}
+    {{- if tpl "{{ .Values.controller.affinity }}" . | contains "nodeSelectorTerms" }}
```

Comment:
I think the logic should be 
```
if or .Values.controller.runOnControlPlane .Values.controller.runOnMaster}}

else
  use Values.controller.affinity
```

Reply:
It may be better to keep it as is for backward compatibility, and because it provides logical fallback from more specific to more generic configuration.
Users can provide both controller.affinity property and controller.runOnControlPlane property, in their values.yaml, and it will work for both older version of the chart, and newer version of the chart with the bug fixed.